lib/lambda_ethereum_consensus/state_transition/misc.ex

```diff
@@ -152,27 +149,29 @@ defmodule LambdaEthereumConsensus.StateTransition.Misc do
     start_ = div(length(indices) * index, count)
     end_ = div(length(indices) * (index + 1), count) - 1
 
-    Enum.reduce(start_..end_, {:ok, []}, fn i, acc ->
-      case acc do
-        {:ok, acc_list} ->
-          case compute_shuffled_index(i, length(indices), seed) do
-            {:ok, shuffled_index} ->
-              {:ok, [Enum.at(indices, shuffled_index) | acc_list]}
-
-            {:error, _} = error ->
-              error
-          end
+    result =
+      Enum.reduce_while(start_..end_, {:ok, []}, fn i, acc ->
```

Comment:
I would move the `Enum.reduce_while` into `compute_committee_indices` since the function is coupled to the concept of a reducer (uses :halt, :cont)